### PR TITLE
[FrameworkBundle] Configurable Serializer name converter

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | symfony/symfony-docs#5483 |
- [x] Add tests

### CODE REVIEW

**Comment by dunglas** [comment]
ping @symfony/deciders

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 1]
```diff
@@ -970,6 +970,12 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
                 1, new Reference($config['cache'])
             );
         }
+
+        if (isset($config['name_converter']) && $config['name_converter']) {
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: IIRC, the `isset` part is not needed.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -970,6 +970,12 @@ private function registerSerializerConfiguration(array $config, ContainerBuilder
...
+            $container->getDefinition('serializer.normalizer.object')->replaceArgument(
+                1, new Reference($config['name_converter'])
+            );
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: should be on one line.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml [hunk 3]
```diff
@@ -20,6 +20,7 @@
         <!-- Normalizer -->
         <service id="serializer.normalizer.object" class="Symfony\Component\Serializer\Normalizer\ObjectNormalizer" public="false">
             <argument type="service" id="serializer.mapping.class_metadata_factory" />
+            <argument>null</argument> <!-- Replaced by the extension if a name converted is configured -->
```
Comment (fabpot) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/serializer.xml]: The comment should only be about what this is about, so something like `<!-- name converted -->`

**Comment by fabpot** [comment]
Can you also create the PR for the docs?

**Comment by xabbuh** [comment]
Don't you have to set the default value of the `name_converter` option to `serializer.name_converter.camel_case_to_snake_case` so that it is taken by default?

**Comment by dunglas** [comment]
No. If you do that everything will be converted by default and this is not intended nor a best practice.

The name converter definition is here just for convenience and will be removed during the container compilation when not used.

**Comment by dunglas** [comment]
Code updated and doc added (symfony/symfony-docs#5483)

**Comment by xabbuh** [comment]
> No. If you do that everything will be converted by default and this is not intended nor a best practice.

> The name converter definition is here just for convenience and will be removed during the container compilation when not used.

Thanks for explaining your intentions.

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request merged**